### PR TITLE
support RTL in toolbar

### PR DESCRIPTION
**Pull request by 1akhanBaheti**
# Feature Preview 

image

FR raised here also :
https://github.com/AppFlowy-IO/AppFlowy/issues/3481

### CODE REVIEW

**Review by MayurSMahajan — APPROVED** [review]
Can you add a test for this, so we ensure new changes don't break your feature.

**Comment by 1akhanBaheti** [comment]
ok @MayurSMahajan, I will write test also for this.
but i have one query, that whether i should write test for toolbar widget  OR **computeActiveItems()** function to check whether it generating toolbarItems with correct order of ToolbarItemID.

**Comment by LucasXu0** [comment]
Nice try!

**Comment by 1akhanBaheti** [comment]
There was commit lint issue, that's why jobs were failing so,
i have refactored the commit message, any other changes required ? @MayurSMahajan

**Review by MayurSMahajan — APPROVED** [review]

**Comment by MayurSMahajan** [comment]
Looks good to me.

File: lib/src/editor/toolbar/desktop/floating_toolbar.dart [hunk 1]
```diff
@@ -168,14 +170,39 @@ class _FloatingToolbarState extends State<FloatingToolbar>
...
+          ? a.group.compareTo(b.group)
+          : b.group.compareTo(a.group),
+    );
+    // insert the divider.
+    return activeItems
+        .splitBetween((first, second) => first.group != second.group)
+        .expand((element) => [...element, placeholderItem])
+        .toList()
+      ..removeLast();
+  }
```
Comment (LucasXu0) [inline lib/src/editor/toolbar/desktop/floating_toolbar.dart:204]: Why do you move this code from floating_toolbar_widget to here?

File: test/new/infra/testable_editor.dart [hunk 2]
```diff
@@ -42,6 +42,7 @@ class TestableEditor {
     Widget Function(Widget child)? wrapper,
     TargetPlatform? platform,
     String? defaultTextDirection,
+    TextDirection toolbarLayoutDirection=TextDirection.ltr,
```
Comment (LucasXu0) [inline test/new/infra/testable_editor.dart:45]:
```suggestion
    TextDirection toolbarLayoutDirection = TextDirection.ltr,
```

**Comment by LucasXu0** [comment]
> There was commit lint issue, that's why jobs were failing so, i have refactored the commit message, any other changes required ? @MayurSMahajan

yes. so please run `flutter analyze`, `dart format .`, and 'flutter test' before submitting your code.

File: lib/src/editor/toolbar/desktop/floating_toolbar.dart (hunk 1 above)
Comment (1akhanBaheti) [reply lib/src/editor/toolbar/desktop/floating_toolbar.dart:204]: actually in tests, I was not able to verify the order of toolbarItems, that's why moved there so now I am able to verify it by taking it from parameters

Comment (1akhanBaheti) [reply lib/src/editor/toolbar/desktop/floating_toolbar.dart:204]: @LucasXu0 
any issue with this approach ?

**Comment by LucasXu0** [comment]
@1akhanBaheti I optimized some parts of your code. please feel free to check this [commit](https://github.com/AppFlowy-IO/appflowy-editor/pull/496/commits/df309683b229c0be22fdca0805f0465f68ff02bd).

the most important change is that I used the text direction of the row instead of reversing the toolbar items.

**Pull request merged**